src/core.js

```diff
@@ -301,245 +301,250 @@ async function pruneNonCriticalCssLauncher ({
   // NOTE: would need a refactor to killTimeout logic to be able to remove promise here.
   /* eslint-disable no-async-promise-executor */
   return new Promise(async (resolve, reject) => {
-    /* eslint-enable no-async-promise-executor */
-    debuglog('Penthouse core start')
-    let page = null
-    let killTimeout = null
-    async function cleanupAndExit ({ error, returnValue }) {
-      if (_hasExited) {
-        return
-      }
-      debuglog('cleanupAndExit')
-      _hasExited = true
-      clearTimeout(killTimeout)
+    // ensure there is no uncaughtException
+    try {
```

Comment:
Can you move the this line (the opening of the try catch block) to be below the cleanupAndExit function definition, i.e just around the code that can actually through exceptions during the runtime?

Reply:
(y)